src/content/containers/TimelineFlameChart.js

```diff
@@ -0,0 +1,147 @@
+   * SMALL GRAPH: The smallest it can be is 1.5 times the row height, giving a visual cue
+   *              to the user that this row is expanded, even if it's super shallow.
+   * LARGE GRAPH: If the flame chart is too large, only expand out to most of the
+   *              available space, leaving some margin to show the other rows.
+   */
+  getViewHeight(maxViewportHeight: number): number {
+    const { viewHeight, isThreadExpanded } = this.props;
+    const exactSize = isThreadExpanded ? maxViewportHeight * 1.5 : maxViewportHeight;
+    const largeGraph = viewHeight - TIMELINE_ROW_HEIGHT * 2;
+    const smallGraph = TIMELINE_ROW_HEIGHT;
+    return Math.max(smallGraph, Math.min(exactSize, largeGraph));
+  }
+
+  /**
+   * Determine
+   */
+  getMaximumZoom(): UnitIntervalOfProfileRange {
+    const {
+      timeRange: { start, end },
+      interval,
+    } = this.props;
+    return interval / (end - start);
+  }
+
+  render() {
+    const {
+      thread, isThreadExpanded, maxStackDepth, stackTimingByDepth, isSelected, timeRange,
+      threadIndex, interval, getCategory, getLabel, horizontalViewport,
+      changeTimelineHorizontalViewport,
+    } = this.props;
+
+    // The viewport needs to know about the height of what it's drawing, calculate
+    // that here at the top level component.
+    const maxViewportHeight = maxStackDepth * STACK_FRAME_HEIGHT;
+    const title = thread.processType ? `${thread.name} [${thread.processType}]` : thread.name;
+    const height = this.getViewHeight(maxViewportHeight);
+    const buttonClass = classNames('timelineFlameChartCollapseButton', {
+      expanded: isThreadExpanded,
+      collapsed: !isThreadExpanded,
+    });
+
+    return (
+      <div className='timelineFlameChart' style={{ height }}>
+        <div className='timelineFlameChartLabels grippy'>
+          <span>{title}</span>
+          <button className={buttonClass} onClick={this.toggleThreadCollapse} />
+        </div>
+        <FlameChartViewport key={threadIndex}
+                            thread={thread}
+                            maxStackDepth={maxStackDepth}
+                            isThreadExpanded={isThreadExpanded}
+                            stackTimingByDepth={stackTimingByDepth}
+                            isSelected={isSelected}
+                            timeRange={timeRange}
+                            threadIndex={threadIndex}
+                            interval={interval}
+                            maxViewportHeight={maxViewportHeight}
+                            stackFrameHeight={STACK_FRAME_HEIGHT}
+                            getCategory={getCategory}
+                            getLabel={getLabel}
+                            maximumZoom={this.getMaximumZoom()}
+                            horizontalViewport={horizontalViewport}
+                            changeTimelineHorizontalViewport={changeTimelineHorizontalViewport }/>
```

Comment:
Wow, this is a lot of props. Not sure if we should do something about that, or what to do about it.